xbmc/windows/GUIWindowScreensaver.cpp

```diff
@@ -68,6 +68,9 @@ bool CGUIWindowScreensaver::OnMessage(CGUIMessage& message)
 
   case GUI_MSG_WINDOW_INIT:
     {
+      // override the clear colour - we must never clear fullscreen
+      m_clearBackground = 0;
```

Comment:
Found another way, the way before with count was really not nice.

There are also a few other places in Kodi that go that way. Also, 3D video rendering must avoid a glClear (again other way) is called.

Reply:
See https://github.com/xbmc/xbmc/pull/15148#discussion_r252569744 - not calling `glClear` does not mean that the buffer contents are preserved.

Reply:
> Why is it then also needed on other fullscreen places in Kodi?

How do you know it is needed? It might just be a performance optimization (no use clearing the buffer if we draw over everything anyway) - do you have a specific place where you know that it is really necessary?

> Does the complete screen content have to be copied with every render call to have it back at the next frame?

Yes and no. There are extensions that can be used instead of the often unsupported `EGL_BUFFER_PRESERVED`. But they will not give you the last frame, but instead an arbitrary previous frame and tell you how old it is (`EGL_EXT_buffer_age`) - this also works with Wayland. But it will not help with the solarwinds screensaver because I guess that it is implemented in a way that it needs the exact previous frame. Then yes, you need to copy when swapping buffers. This is also what the driver would usually do for you anyway when setting `EGL_BUFFER_PRESERVED`. See also e.g. https://community.arm.com/graphics/b/blog/posts/mali-performance-3-is-egl_5f00_buffer_5f00_preserved-a-good-thing

Other option is to modify solarwinds screensaver to always draw everything.